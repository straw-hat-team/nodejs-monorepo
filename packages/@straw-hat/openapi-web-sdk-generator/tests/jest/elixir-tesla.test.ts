import { createTmpDir, readPetStoreSpec } from './support-files';
import * as path from 'path';
import ElixirTeslaCodegen from '../../src/generators/elixir-tesla';

test('elixir-tesla generator', async () => {
  // GIVEN
  const tmpDir = await createTmpDir(['elixir-tesla']);
  const openapiDocument = await readPetStoreSpec();

  const generator = new ElixirTeslaCodegen({
    outputDir: tmpDir.path,
  }).setDocument(openapiDocument);

  // WHEN
  await generator.generate();

  // THEN
  for await (const filePath of tmpDir.walkFiles('.')) {
    const snapshotName = path.relative(tmpDir.path, filePath);
    expect(await tmpDir.readFile(filePath)).toMatchSnapshot(snapshotName);
  }
});
