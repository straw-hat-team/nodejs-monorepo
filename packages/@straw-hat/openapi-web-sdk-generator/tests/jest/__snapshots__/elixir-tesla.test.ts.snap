// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elixir-tesla generator: api.ex 1`] = `
"
       def update_pet()
         :ok
       end
      
       def add_pet()
         :ok
       end
      
       def find_pets_by_status()
         :ok
       end
      
       def find_pets_by_tags()
         :ok
       end
      
       def get_pet_by_id()
         :ok
       end
      
       def update_pet_with_form()
         :ok
       end
      
       def delete_pet()
         :ok
       end
      
       def upload_file()
         :ok
       end
      
       def get_inventory()
         :ok
       end
      
       def place_order()
         :ok
       end
      
       def get_order_by_id()
         :ok
       end
      
       def delete_order()
         :ok
       end
      
       def create_user()
         :ok
       end
      
       def create_users_with_list_input()
         :ok
       end
      
       def login_user()
         :ok
       end
      
       def logout_user()
         :ok
       end
      
       def get_user_by_name()
         :ok
       end
      
       def update_user()
         :ok
       end
      
       def delete_user()
         :ok
       end
      "
`;
